### DISCUSSION TITLE
ike sa handling on receiving ike-rekey event

### DISCUSSION THREAD

**Discussion by hbhokare** [strongswan/strongswan#561] [Q&A]
Hi,
We are trying to develop the IPSec-based application, using different strongswan/davici API from our application. We have below questions related to the handling of ike-sa object in the application during the ike-rekey event
 
1. Do we expect the ike-down event with older ike-sa unique-id after we get ike-rekey event 
2.  if the answer to the first question is no then I can only update ike-sa object allocated in the application with the new unique-id got from the ike-rekey event assuming that any further communication of the ike-up-down event would happen from strongswan using this updated unique id .

Thanks,
-Harish

**Comment by tobiasbrunner** [comment]
> Do we expect the ike-down event with older ike-sa unique-id after we get ike-rekey event

No, _ike-updown_ is not called for any IKE_SAs involved in a rekeying.

>  if the answer to the first question is no then I can only update ike-sa object allocated in the application with the new unique-id got from the ike-rekey event assuming that any further communication of the ike-up-down event would happen from strongswan using this updated unique id .

Correct, you have to handle the _ike-rekey_ event to track IKE_SAs over rekeyings.

**Reply by hbhokare** [reply]
ok thanks !
